### PR TITLE
[ONNX] Update arange op to support out argument

### PR DESCRIPTION
**Pull request by shubhambhokare1**
Update arange op to support out argument

### CODE REVIEW

File: torch/onnx/symbolic_helper.py [hunk 1]
```diff
@@ -376,7 +376,7 @@ def _is_all_integral(scalars):
     # infer input types from dtype. If not, then check if any of start, stop,
     # or step are floating point, and infer the type from get_default.
     # Otherwise, the dtype is inferred to be torch.int64.
-    if _is_value(dtype) and _is_none(dtype):
+    if not _is_value(dtype) and dtype is None:
```
Comment (KsenijaS) [inline torch/onnx/symbolic_helper.py:379]: I'm not sure why check for _is_value needed, if dtype is not None it will go to else block

Comment (shubhambhokare1) [reply torch/onnx/symbolic_helper.py:379]: Fixed this, the _get_arange_dtype helper produces a torch.C._Value, for cases in which dtype is not provided, hence the check

File: torch/onnx/symbolic_opset11.py [hunk 2]
```diff
@@ -487,7 +487,17 @@ def _get_arange_dtype(dtype):
         dtype = sym_help._maybe_get_const(dtype, 'i')
         return dtype
 
-    if len(args) == 5:
+    if len(args) == 2:
```
Comment (neginraoof) [inline torch/onnx/symbolic_opset11.py:490]: If the logic in len(args)==2 branch is very close to logic in len(args)==5 branch, can you see if you can refactor the code to remove duplicate lines?
Same for the other branches that can be merged.

File: torch/onnx/symbolic_opset11.py [hunk 3]
```diff
@@ -487,7 +487,17 @@ def _get_arange_dtype(dtype):
...
+        # aten::arange(Scalar end, Tensor out)
+        type, end, start, step = sym_help._arange_cast_helper(g, end=args[0])
+        start_default = g.op("Constant", value_t=torch.tensor(0, dtype=sym_help.scalar_type_to_pytorch_type[type]))
+        delta_default = g.op("Constant", value_t=torch.tensor(1, dtype=sym_help.scalar_type_to_pytorch_type[type]))
+        arange_tensor = g.op("Range", start_default, end, delta_default)
+    elif len(args) == 4:
```
Comment (neginraoof) [inline torch/onnx/symbolic_opset11.py:496]: What if len(args) == 3?
Does it happen? Like 
aten::arange(Scalar start, Scalar end, Tensor out)

File: torch/onnx/symbolic_opset9.py [hunk 4]
```diff
@@ -2249,7 +2249,23 @@ def _get_arange_dtype(dtype):
...
         return dtype
 
-    if len(args) == 5:
+    if len(args) == 2:
+        # aten::arange(Scalar end, Tensor out)
+        end = g.op("Unsqueeze", args[0], axes_i=[0])
+        dtype = 4  # default to int64
+        arange_tensor = g.op("Squeeze", nonzero(g, ones(g, end, dtype, None, None)), axes_i=[1])
+        return g.op("Cast", arange_tensor, to_i=sym_help.scalar_type_to_onnx[dtype])
+    elif len(args) == 4:
```
Comment (neginraoof) [inline torch/onnx/symbolic_opset9.py:2258]: Same comments as in opset 11.

File: torch/onnx/symbolic_opset11.py (hunk 3 above)
Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset11.py:496]: Arange signature for 3 args doesn't exist in https://github.com/pytorch/pytorch/blob/348e78b08600e4652e57ed892ba0cf9c7a1661a0/aten/src/ATen/native/native_functions.yaml

File: torch/onnx/symbolic_opset11.py (hunk 2 above)
Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset11.py:490]: Made necessary refactoring changes

File: torch/onnx/symbolic_opset9.py (hunk 4 above)
Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset9.py:2258]: Only one/three lines are duplicated in opset 9, would it still be necessary to refactor?

File: torch/onnx/symbolic_helper.py [hunk 5]
```diff
@@ -380,7 +380,7 @@ def _is_all_integral(scalars):
     # infer input types from dtype. If not, then check if any of start, stop,
     # or step are floating point, and infer the type from get_default.
     # Otherwise, the dtype is inferred to be torch.int64.
-    if _is_value(dtype) and _is_none(dtype):
+    if (_is_value(dtype) and _is_none(dtype)) or dtype is None:
```
Comment (BowenBao) [inline torch/onnx/symbolic_helper.py:383]: nit: consider moving `dtype is None` to the front for short circuiting.

**Review by neginraoof — APPROVED** [review]
Thanks!

**Review by bzinodev — APPROVED** [review]

**Pull request closed without merging**